src/main/java/ch/njol/skript/util/BlockUtils.java

```diff
@@ -99,4 +101,23 @@ public static Location getLocation(final @Nullable Block b) {
 		return l;
 	}
 	
+	@Nullable
+	public static BlockData createBlockData(String dataString) {
+		// Skript uses a comma to separate lists, so we use a semi colon as a delimiter
+		// Here we are just replacing it back to a comma to create a new block data
+		String data = dataString.replace(";", ",");
+		// remove white space within square brackets ([ lit = false] -> [lit=false])
+		data = data.replaceAll(" (?=[^\\[]*])", "");
+		// if there are spaces before the square bracket we remove that
+		data = data.replace(" [", "[");
+		// and replace white space between namespace with underscores
```

Comment:
Minor nitpick, but the first and second comment lines are properly capitalized, but then the subsequent ones aren't

Reply:
wHaT iS wRoNg WiTh ThAt?
Im joking haha! Yeah, ill fix it up :)

Reply:
Ok, the casings have been fixed 👍